### PR TITLE
RavenDB-21628 Move Connection Strings to React

### PR DESCRIPTION
**Pull request by kalczur**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21628/Move-Connection-Strings-to-React

### Type of change

- New feature

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

image

### CODE REVIEW

**Comment by ppekrol** [comment]
Status?

**Comment by kalczur** [comment]
> Status?

Waiting for review

**Comment by ml054** [comment]
@kalczur Please rebase on top of latest v6.0 and we will start review/ testing.

**Comment by kalczur** [comment]
@ml054 Done

File: src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                            )}
+                        </div>
+                    )}
+                </Collapse>
+            </CardBody>
+        </Card>
+    );
+}
+
+const fieldBase = "destinations.glacier";
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx:155]: should we pass this from external ?

File: src/Raven.Studio/typescript/components/common/formDestinations/AmazonS3.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                            )}
+                        </>
+                    )}
+                </Collapse>
+            </CardBody>
+        </Card>
+    );
+}
+
+const fieldBase = "destinations.s3";
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/formDestinations/AmazonS3.tsx:229]: same here + assume the other connection destinations

File: src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx (hunk 1 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx:155]: Maybe, but it doesn't really matter. If I made a typo here, there would be an error in every use of getName().

Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx:155]: additionally, it is directly linked to this component and never changes

Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/formDestinations/AmazonGlacier.tsx:155]: if we want to use it in another place, I assume that we must have the `destinations` field there too

**Comment by ml054** [comment]
@ppekrol We want to test that with real data before review. Except that LGTM.

**Review by ml054 — APPROVED** [review]

**Pull request merged**